modules/representation_management/app/services/representation_management/accredited_entity_query.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RepresentationManagement
+  class AccreditedEntityQuery
+    MAXIMUM_RESULT_COUNT = 10
+    WORD_SIMILARITY_THRESHOLD = 0.7
+
+    def initialize(query_string)
+      @query_string = query_string
+    end
+
+    def results
+      return [] if @query_string.blank?
+
+      array_results = ActiveRecord::Base.connection.exec_query(
+        ActiveRecord::Base.send(:sanitize_sql_array, [
+                                  sql_query,
+                                  {
+                                    query_string: @query_string,
+                                    threshold: WORD_SIMILARITY_THRESHOLD,
+                                    max_results: MAXIMUM_RESULT_COUNT
+                                  }
+                                ])
+      )
+
+      transform_results_to_objects(array_results)
+    end
+
+    private
+
+    # rubocop:disable Metrics/MethodLength
+    def sql_query
+      <<-SQL.squish
+        WITH combined AS (
+          SELECT
+            id,
+            full_name AS name,
+            'AccreditedIndividual' AS model_type,
+            levenshtein(full_name, :query_string) AS distance
```

Comment:
Great work! I noticed the use of `levenshtein` and `word_similarity` SQL functions—I'm guessing that's why you opted for raw SQL over ActiveRecord.

The `results` function is well-tested—an excellent job of ensuring thorough coverage.

A couple of suggestions: The `sql_query` method could benefit from being broken down into smaller methods to enhance readability. If breaking it down isn't preferred, perhaps a more descriptive name for the `sql_query` method could clarify its purpose. Additionally, some method documentation would be helpful. Consider adding comments like:
```
# This scope identifies individuals whose full name matches the query string.
# We use Levenshtein distance to rank results by similarity...
# The word_similarity function applies a threshold to filter out less relevant matches...
```